### PR TITLE
Fix #8921 - SQL file report for Output:Table:Monthly ColumnName field missing closing curly brace

### LINKED ISSUE #8921: SQL file report for Output:Table:Monthly ColumnName field missing closing curly brace
Issue overview
------------
@JasonGlazer 
Minor issue -
The following Output:Table:Monthly with 'Maximum' aggregation type will produce a table in the HTML output report with the first column header "ELECTRICITY:FACILITY {Maximum}[W]", but in the sql file the ColumnName for that field is missing the closing curly brace "ELECTRICITY:FACILITY {Maximum".

```
Output:Table:Monthly,
  Building Energy Performance - Electricity Peak Demand, !- Name
  2,                                      !- Digits After Decimal
  Electricity:Facility,                   !- Variable or Meter Name 1
  Maximum,                                !- Aggregation Type for Variable or Meter 1
  InteriorLights:Electricity,             !- Variable or Meter Name 2
  ValueWhenMaximumOrMinimum,              !- Aggregation Type for Variable or Meter 2
  ExteriorLights:Electricity,             !- Variable or Meter Name 3
  ValueWhenMaximumOrMinimum,              !- Aggregation Type for Variable or Meter 3
  InteriorEquipment:Electricity,          !- Variable or Meter Name 4
  ValueWhenMaximumOrMinimum,              !- Aggregation Type for Variable or Meter 4
  ExteriorEquipment:Electricity,          !- Variable or Meter Name 5
  ValueWhenMaximumOrMinimum,              !- Aggregation Type for Variable or Meter 5
  Fans:Electricity,                       !- Variable or Meter Name 6
  ValueWhenMaximumOrMinimum,              !- Aggregation Type for Variable or Meter 6
  Pumps:Electricity,                      !- Variable or Meter Name 7
  ValueWhenMaximumOrMinimum,              !- Aggregation Type for Variable or Meter 7
  Heating:Electricity,                    !- Variable or Meter Name 8
  ValueWhenMaximumOrMinimum,              !- Aggregation Type for Variable or Meter 8
  Cooling:Electricity,                    !- Variable or Meter Name 9
  ValueWhenMaximumOrMinimum,              !- Aggregation Type for Variable or Meter 9
  HeatRejection:Electricity,              !- Variable or Meter Name 10
  ValueWhenMaximumOrMinimum,              !- Aggregation Type for Variable or Meter 10
  Humidifier:Electricity,                 !- Variable or Meter Name 11
  ValueWhenMaximumOrMinimum,              !- Aggregation Type for Variable or Meter 11
  HeatRecovery:Electricity,               !- Variable or Meter Name 12
  ValueWhenMaximumOrMinimum,              !- Aggregation Type for Variable or Meter 12
  WaterSystems:Electricity,               !- Variable or Meter Name 13
  ValueWhenMaximumOrMinimum,              !- Aggregation Type for Variable or Meter 13
  Cogeneration:Electricity,               !- Variable or Meter Name 14
  ValueWhenMaximumOrMinimum;              !- Aggregation Type for Variable or Meter 14
```

image

image

### Details
Some additional details for this issue (if relevant):
 - Windows 10
 - Observed in EnergyPlus v9.2.0 and v9.5.0

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

Fix #8921 - SQL file report for Output:Table:Monthly ColumnName field missing closing curly brace

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [x] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [x] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/OutputReportTabular.cc [hunk 1]
```diff
@@ -7379,7 +7379,7 @@ void WriteMonthlyTables(EnergyPlusData &state)
                                 curUnits = "ton";
                                 curConversionFactor *= 3600.0;
                             }
-                            columnHead(columnRecount - 1) = ort->MonthlyColumns(curCol).varName + curAggString + '[' + curUnits + ']';
+                            columnHead(columnRecount - 1) = ort->MonthlyColumns(curCol).varName + curAggString + " [" + curUnits + ']';
```
Comment (jmarrec) [inline src/EnergyPlus/OutputReportTabular.cc:7382]: Fix here

File: src/EnergyPlus/OutputReportTabularAnnual.cc [hunk 2]
```diff
@@ -832,7 +832,7 @@ void AnnualTable::writeTable(EnergyPlusData &state, OutputReportTabular::iUnitsS
                 curUnits += "/s";
             }
             fixUnitsPerSecond(curUnits, curConversionFactor);
-            columnHead(columnRecount - 1) = fldStIt->m_colHead + curAggString + '[' + curUnits + ']';
+            columnHead(columnRecount - 1) = fldStIt->m_colHead + curAggString + " [" + curUnits + ']';
```
Comment (jmarrec) [inline src/EnergyPlus/OutputReportTabularAnnual.cc:835]: here

File: src/EnergyPlus/OutputReportTabularAnnual.cc [hunk 3]
```diff
@@ -1294,7 +1294,7 @@ void AnnualTable::convertUnitForDeferredResults(EnergyPlusData &state,
     Real64 energyUnitsConversionFactor = AnnualTable::setEnergyUnitStringAndFactor(unitsStyle, energyUnitsString);
     // do the unit conversions
     if (unitsStyle == OutputReportTabular::iUnitsStyle::InchPound) {
-        varNameWithUnits = fldStIt->m_variMeter + '[' + unitEnumToString(fldStIt->m_varUnits) + ']';
+        varNameWithUnits = fldStIt->m_variMeter + " [" + unitEnumToString(fldStIt->m_varUnits) + ']';
```
Comment (jmarrec) [inline src/EnergyPlus/OutputReportTabularAnnual.cc:1297]: And here

File: src/Transition/OutputRulesFiles/OutputChanges9-5-0-to-9-6-0.md [hunk 4]
```diff
@@ -61,4 +61,28 @@ See PR [#8891](https://github.com/NREL/EnergyPlus/pull/8891).
...
+ ELECTRICITY:FACILITY {TIMESTAMP}
+```
+
+And in the SQL file, `ColumnName` will as a result change as follows:
+
+```diff
+-ELECTRICITY:FACILITY {MAXIMUM
++ELECTRICITY:FACILITY {MAXMIUM}
+ ELECTRICITY:FACILITY {TIMESTAMP}
+```
```
Comment (jmarrec) [inline src/Transition/OutputRulesFiles/OutputChanges9-5-0-to-9-6-0.md:88]: Document the output changes

File: tst/EnergyPlus/unit/OutputReportTabular.unit.cc [hunk 5]
```diff
@@ -10048,3 +10048,87 @@ TEST_F(EnergyPlusFixture, OutputReportTabularTest_PredefinedTable_Standard62_1_W
...
+        )", "TabularDataWithStrings"
+    );
+    state->dataSQLiteProcedures->sqlite->sqliteCommit();
+
+    // Should be none!
+    for (auto& col: missingBracesHeaders) {
+        std::string colHeader = col[0];
+        EXPECT_TRUE(false) << "Missing braces in monthly table for : " << colHeader;
+    }
+}
```
Comment (jmarrec) [inline tst/EnergyPlus/unit/OutputReportTabular.unit.cc]: Test

File: tst/EnergyPlus/unit/OutputReportTabularAnnual.unit.cc [hunk 6]
```diff
@@ -453,3 +455,106 @@ TEST_F(EnergyPlusFixture, OutputReportTabularAnnual_invalidAggregationOrder)
...
+        )", "TabularDataWithStrings"
+    );
+    state->dataSQLiteProcedures->sqlite->sqliteCommit();
+
+    // Should be none!
+    for (auto& col: missingBracesHeaders) {
+        std::string colHeader = col[0];
+        EXPECT_TRUE(false) << "Missing braces in monthly table for : " << colHeader;
+    }
+}
```
Comment (jmarrec) [inline tst/EnergyPlus/unit/OutputReportTabularAnnual.unit.cc]: And Test for Output:Table:Annual since I realized it had a similar issue

**Comment by Myoldmopar** [comment]
This seems to be in very good shape, but CI didn't run much on it.  I'm going to pull develop in and let it pass over it today before merging.

**Comment by Myoldmopar** [comment]
Tested sufficiently locally, seems just fine.  Merging.  Thanks @jmarrec

**Pull request merged**